### PR TITLE
tests/imap-detection: add simple test for protocol detection - v6

### PR DESCRIPTION
**Pull request by mmaatuq**
## Ticket #2886

Redmine ticket: [imap protocol detection is incomplete](https://redmine.openinfosecfoundation.org/issues/2886)
Relates to this [PR](https://github.com/OISF/suricata/pull/11219)
changes from PR #1819 :  rebase to latest master.

### CODE REVIEW

**Review by catenacyber — APPROVED** [review]
SV test looks good to me to show imap detection

File: tests/imap-detection/test.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      proto: TCP
+      src_ip: 131.151.32.21
+      src_port: 4167
+      tcp.ack: true
+      tcp.fin: true
+      tcp.psh: true
+      tcp.state: closed
+      tcp.syn: true
+      tcp.tc_max_regions: 1
+      tcp.tcp_flags: 1b
```
Comment (catenacyber) [inline tests/imap-detection/test.yaml:32]: Nit : we do not need to test all these details

Comment (mmaatuq) [reply tests/imap-detection/test.yaml:32]: > Nit : we do not need to test all these details

should i also remove all tcp-related details? 
like 
```C
-      tcp.ack: true
-      tcp.fin: true
-      tcp.psh: true
-      tcp.state: closed
-      tcp.syn: true
-      tcp.tc_max_regions: 1
-      tcp.tcp_flags: 1b
-      tcp.tcp_flags_tc: 1b
-      tcp.tcp_flags_ts: 1b
-      tcp.ts_max_regions: 1
```

Comment (catenacyber) [reply tests/imap-detection/test.yaml:32]: indeed

**Pull request closed without merging**